core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -604,14 +604,14 @@ class AdminManager(val config: KafkaConfig,
         case OpType.APPEND => {
           if (!listType(alterConfigOp.configEntry().name(), configKeys))
             throw new InvalidRequestException(s"Config value append is not allowed for config key: ${alterConfigOp.configEntry().name()}")
-          val oldValueList = configProps.getProperty(alterConfigOp.configEntry().name()).split(",").toList
+          val oldValueList = configProps.getProperty(alterConfigOp.configEntry().name(), "").split(",").toList.filter(_.trim.length > 0)
```

Comment:
The filter code is necessary to generate an empty list when there's no value.

Reply:
Do we need the `trim` call? We should already get a length 0 string back if the property is empty. Seems unlikely that we'd ever have a list value that includes a whitespace-only string as an element, but who knows.

Reply:
I'll remove and try.

Reply:
Removed the trim and it still passes tests. yay.